src/pages/AuthorsForm.vue

```diff
@@ -33,24 +49,52 @@ export default {
   data () {
     return {
       values: {
+        id: '',
         name: '',
         email: ''
-      }
+      },
+      isCreateMode: true,
+      confirm: false
     }
   },
 
+  async created () {
+    this.values.id = this.$route.params.id
+    this.isCreateMode = this.$route.name !== 'AuthorsEdit'
+    await this.fetchAuthors()
+    if (this.values.id) this.retrieveAuthorById(this.values.id)
+  },
```

Comment:
Alguns pontos aqui:
- Temos costume de validar se é pagina de edição pelo ID não pelo nome, ou seja, se tem ID é edição
- Qual a necessidade de vc setar o `this.values.id` sendo que vc n precisa enviar pro payload isso??
- Se a request `fetchAuthors()` der erro?